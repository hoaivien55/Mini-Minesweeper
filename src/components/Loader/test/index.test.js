import renderer from 'react-test-renderer';
import { render } from "@testing-library/react";
import Loader from "..";


it('Loader: Virtual DOM snapshot', () => {
    const tree = renderer
      .create(<Loader show/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

test("Loader: not show loader", () => {
    render(<Loader show={false}/>);
    const divElement = document.querySelector(".lds-ripple");
    expect(divElement).toBeNull()
})