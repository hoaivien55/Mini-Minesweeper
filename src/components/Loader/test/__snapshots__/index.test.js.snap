// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader: Virtual DOM snapshot 1`] = `
.c0 {
  position: absolute;
  height: 100%;
  width: 100%;
  background: #8080804d;
  top: 0;
  left: 0;
  border-radius: 12px;
}

.c0 .lds-ripple {
  display: inline-block;
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 80px;
  height: 80px;
}

.c0 .lds-ripple div {
  position: absolute;
  border: 4px solid #fff;
  opacity: 1;
  border-radius: 50%;
  -webkit-animation: lds-ripple 1s cubic-bezier(0,0.2,0.8,1) infinite;
  animation: lds-ripple 1s cubic-bezier(0,0.2,0.8,1) infinite;
}

.c0 .lds-ripple div:nth-child(2) {
  -webkit-animation-delay: -0.5s;
  animation-delay: -0.5s;
}

<div
  className="c0"
>
  <div
    className="lds-ripple"
  >
    <div />
    <div />
  </div>
</div>
`;
